plugins/otter-pro/inc/plugins/class-form-pro-features.php

```diff
@@ -347,6 +349,150 @@ public function send_autoresponder( $form_data ) {
 		}
 	}
 
+	/**
+	 * Send autoresponder email to the subscriber.
+	 *
+	 * @param Form_Data_Request|null $form_data The files to load.
+	 * @since 2.4
+	 */
+	public function trigger_webhook( $form_data ) {
+
+		if ( ! isset( $form_data ) ) {
+			return $form_data;
+		}
+
+		if (
+			( ! class_exists( 'ThemeIsle\GutenbergBlocks\Integration\Form_Data_Request' ) ) ||
+			! ( $form_data instanceof \ThemeIsle\GutenbergBlocks\Integration\Form_Data_Request ) ||
+			$form_data->has_error() ||
+			empty( $form_data->get_form_options()->get_webhook_id() )
+		) {
+			return $form_data;
+		}
+
+		try {
+			$form_webhook_id = $form_data->get_form_options()->get_webhook_id();
+
+			$webhooks = get_option( 'themeisle_webhooks_options', array() );
+
+			$webhook = null;
+
+			foreach ( $webhooks as $hook ) {
+				if ( $hook['id'] === $form_webhook_id ) {
+					$webhook = $hook;
+					break;
+				}
+			}
+
+			if ( ! empty( $webhook ) && ! empty( $webhook['url'] ) ) {
+				$method        = empty( $webhook['method'] ) ? 'POST' : $webhook['method'];
+				$url           = $webhook['url'];
+				$headers_pairs = empty( $webhook['headers'] ) ? array() : $webhook['headers'];
+				$headers       = array();
+
+				foreach ( $headers_pairs as $pair ) {
+					if ( empty( $pair['key'] ) || empty( $pair['value'] ) ) {
+						continue;
+					}
+
+					$headers[] = $pair['key'] . ': ' . $pair['value'];
+				}
+
+				$payload = apply_filters( 'themeisle_blocks_form_webhook_payload', array(), $form_data, $webhook );
```

Comment:
A NIT: we can keep the same naming for filters as we do for rest of the code: https://github.com/Codeinwp/otter-blocks/blob/ce56737db046cb5d2bfad6af8656dec8fe1641be/inc/server/class-form-server.php#L241